hypervisor/common/hypercall.c

```diff
@@ -254,15 +254,17 @@ int32_t hcall_create_vm(struct acrn_vcpu *vcpu, struct acrn_vm *target_vm, uint6
 	int32_t ret = -1;
 	struct acrn_vm *tgt_vm = NULL;
 	struct acrn_vm_creation cv;
-	struct acrn_vm_config* vm_config = NULL;
+	struct acrn_vm_config *vm_config = get_vm_config(vmid);
 
 	if (copy_from_gpa(vm, &cv, param1, sizeof(cv)) == 0) {
 		if (is_poweroff_vm(get_vm_from_vmid(vmid))) {
 
-			vm_config = get_vm_config(vmid);
-
 			/* Filter out the bits should not set by DM and then assign it to guest_flags */
-			vm_config->guest_flags |= (cv.vm_flag & DM_OWNED_GUEST_FLAG_MASK);
+			if (is_static_vm_configured(target_vm)) {
+				vm_config->guest_flags |= (cv.vm_flag & DM_OWNED_GUEST_FLAG_MASK);
+			} else {
+				vm_config->guest_flags = (cv.vm_flag & DM_OWNED_GUEST_FLAG_MASK);
+			}
```

Comment:
WHy we need this change? I think We don't need.  And the static/dynamic flag should be a HV owned bits.

Reply:
here copy the guest_flags passed by DM to vm_config,  
for static VM, it keeps the original implementation, it will "OR" with the guest_flags in static vm config
for dynamic VM, we only copy guest_flags  passed by DM to vm_config,  no need to "OR" operation because it will affect the guest_flags for the previous one.

Reply:
I don't think so.  @jsun26intel  can you explain?